sound/soc/sof/intel/hda.c

```diff
@@ -736,6 +736,9 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 	if (sdev->hda)
 		chip = sdev->hda->desc;
 
+	/* codec removal, invoke bus_device_remove */
+	snd_hdac_ext_bus_device_remove(bus);
```

Comment:
almost good, but we want the snd_hdac_ext functions to be selected only when HDA_LINK is enabled. This is not needed when you don't need HDMI or HDAudio codecs.